erpnext/selling/doctype/quoting_sheet/quoting_sheet.js

```diff
@@ -50,6 +76,13 @@ frappe.ui.form.on("Quoting Sheet Item", {
 		if (row.qty && row.rate) {
 			let amount = row.qty * row.rate;
 			frappe.model.set_value(cdt, cdn, "amount", amount);
+			frappe.call({
+				method: "calculate_total_raw_material_cost",
+				doc: frm.doc,
+				callback: () => {
+					frm.refresh_field("rm_cost");
+				}
+			});
```

Comment:
line 64 to 70 and line 79 to 85 contain redundant code can you move it to a common function.